agent/csi/libstorage/csi_service_libstorage.go

```diff
@@ -361,15 +429,141 @@ func (d *driver) NodePublishVolume(
 			"invalid volume capability request"), nil
 	}
 
-	return nil, nil
+	if mv := req.VolumeCapability.GetMount(); mv != nil {
```

Comment:
Hi @codenrhoden,

Might I suggest refactoring this and the next, long, indented logic block into two functions? It may help with debugging in the future and certainly with readability.

Reply:
Not a problem. That's actually how I have it set up in `csi-blockdevices`, so it's a really easy change.